src/request-pipeline/file-request/index.ts

```diff
@@ -47,17 +39,17 @@ export default class FileRequest extends EventEmitter {
     }
 
     private _onError (err: Error) {
-        this.emit('fatalError', getText(MESSAGE.cantReadFile, this.url, err.message));
+        this.emit('fatalError', getText(MESSAGE.cantReadFile, this._url, err.message));
     }
 
-    private _onOpen () {
-        let stream = fs.createReadStream(this.path);
+    private _onOpen (contentStream: any) {
+        let stream = contentStream;
 
         stream = Object.assign(stream, {
             statusCode: 200,
             trailers:   {},
             headers:    {
-                'content-type': mime.lookup(this.path)
+                'content-type': mime.lookup(this._path)
```

Comment:
Could we write a test asserting that we correctly determine MIME types of resources from ASAR?